ostd/src/mm/kspace/mod.rs

```diff
@@ -108,28 +103,58 @@ pub fn paddr_to_vaddr(pa: Paddr) -> usize {
     pa + LINEAR_MAPPING_BASE_VADDR
 }
 
-/// Returns whether the given address should be mapped as tracked.
+/// The kernel page table instance.
 ///
-/// About what is tracked mapping, see [`crate::mm::frame::meta::MapTrackingStatus`].
-pub(crate) fn should_map_as_tracked(addr: Vaddr) -> bool {
-    !(LINEAR_MAPPING_VADDR_RANGE.contains(&addr) || VMALLOC_VADDR_RANGE.contains(&addr))
-}
+/// It manages the kernel mapping of all address spaces by sharing the kernel part. And it
+/// is unlikely to be activated.
+pub static KERNEL_PAGE_TABLE: Once<PageTable<KernelPtConfig>> = Once::new();
 
 #[derive(Clone, Debug)]
 pub(crate) struct KernelPtConfig {}
 
-impl PageTableConfig for KernelPtConfig {
+// We use the first available PTE bit to mark the frame as tracked.
+// SAFETY: `item_into_raw` and `item_from_raw` are implemented correctly,
+unsafe impl PageTableConfig for KernelPtConfig {
     const TOP_LEVEL_INDEX_RANGE: Range<usize> = 256..512;
 
     type E = PageTableEntry;
     type C = PagingConsts;
+
+    type Item = MappedItem;
+
+    fn item_into_raw(item: Self::Item) -> (Paddr, PagingLevel, PageProperty) {
+        match item {
+            MappedItem::Tracked(frame, mut prop) => {
+                debug_assert!(!prop.flags.contains(PageFlags::AVAIL1));
+                prop.flags |= PageFlags::AVAIL1;
+                let level = frame.map_level();
+                let paddr = frame.into_raw();
+                (paddr, level, prop)
+            }
+            MappedItem::Untracked(pa, level, prop) => {
+                debug_assert!(!prop.flags.contains(PageFlags::AVAIL1));
```

Comment:
`prop.flags -= PageFlags::AVAIL1`

Having `AVAIL1` will make it unsound, so we must remove the flag.